arangod/IResearch/IResearchViewBlock.cpp

```diff
@@ -494,42 +581,42 @@ bool IResearchViewUnorderedBlock::next(
   };
```

Comment:
I think the above will copy the entire documents out even for MMFiles, whereas in MMFiles you may get away without copying. I think it would be possible to do
```
if (mmfiles) {
  ctx.res->emplaceValue(ctx.pos, ctx.curRegs, AqlValueHintDocumentNoCopy(doc));
} else {
  ctx.res->emplaceValue(ctx.pos, ctx.curRegs, AqlValueHintCopy(doc));
}
```
You probably don't want to put the if..then..else into the callback, but probably a specific callback per storage engine.